src/twisted/trial/_asynctest.py

```diff
@@ -185,16 +184,16 @@ def _ebDeferTestMethod(self, f, result):
         else:
             result.addError(self, f)
 
-    def deferTearDown(self, ignored, result):
-        d = self._run("tearDown", result)
-        d.addErrback(self._ebDeferTearDown, result)
-        return d
-
-    def _ebDeferTearDown(self, failure, result):
-        result.addError(self, failure)
-        if failure.check(KeyboardInterrupt):
+    async def deferTearDown(self, ignored, result):
```

Comment:
Since we are changing the return value, 
I think that is best to make the "new method" private.

```suggestion
    async def _deferTearDown(self, result):
```

Reply:
Fixed now.